### ISSUE TITLE
1.4.1 G183 F73 - If the text within a link contains the link target, this should not be considered a failure. 

### ISSUE THREAD

**Issue by sebastiannielsen** [w3c/wcag#4143]
Suggestion for update to WCAG guidelines:

If a link text, contains the link target, ergo, the link satisfies one of these conditions:
a link is `href=".*\/(.*)\?.*"` and the text between `<a>` and `</a>` *CONTAINS* the text in (.*), the link is ignored for this rule execution.

a link is `href="mailto:(.*)\?.*"` and the text between `<a>` and `</a>` *CONTAINS* the text in (.*), the link is ignored for this rule execution.

a link is `href="tel:([0123456789+]*)"` and the text between `<a>` and `</a>` evaluated as `/[0123456789+]*/` *CONTAINS* the text in (.*), the link is ignored for this rule execution. (With evaluated as (regexp) I mean that all characters between `<a>` and `</a>` that are not matching the regexp is deleted before comparing, this so a phone number of tel:+1555123123 can be visually displayed as "Call us: +1 (555) 123 123" for example)

so:
Examples of links PASSING G183/F73 regardless of style:
`<a href="mailto:info@example.org">Send us a email on info@example.org!</a>`
`<a href="tel:+1555123123">Call us: +1 (555) 123 123</a>`
`<a href="https://www.example.org">Come visit our partner at www.example.org!</a>`
`<a href="https://www.example.org?referrer=ourCorporation&metrics=topAd">Come visit our partner at www.example.org!</a>`
`<p>There are many resources to find out more about the <a href="rain-in-spain.html">Rain in spain (rain-in-spain.html)</a>.</p>`

Examples of links failing F73 (ergo NOT PASSING if style not fulfilling 3:1 contract ratio to surrounding text or contains other text decoration):
`<a href="mailto:info@example.org">Send us a email here!</a>`
`<a href="tel:+1555123123">Call us now!</a>`
`<a href="https://www.example.org">Come visit our partner Example LTD</a>`

Why?
**I personally think that a link containing its literal technical target, is "other visual indication (besides color) that it is a link."**

With "technical target" I mean that saying "Come visit our parter Example Corp LTD" doesn't tell the user that its a link to "Example Corp LTD's" homepage. BUT, if the link instead says "Come visit our partner www.example.org!" it can be assumed that this is a link to www.example.org . Even if its not, nothing bad happens if a disabled users tries to click that text and it will not hinder or impair the user in using the web content.

A vision impaired user with insufficent color vision, who have trouble finding links in a block of text, should be able to discern web adresses, phone numbers and email adresses, in a block of text. A vision impaired user can try clicking such a area on the screen, and if its a link, it will succeed. If its not a link, nothing bad happens and the user then has to type the URL, email adress or phone number manually.

As compared to other links, it doesn't become a search for a needle in a haystack, to find such links, as they are rare, and it can be assumed that a URL, email adress or phone number, in a web page, is clickable, and always err on the safe side (ergo, try clicking it and see what happens).

I therefore, think that disabled users should have no trouble finding such links EVEN if the style were totally identical to the sourroung text.
Meaning, embedding the link target inside a link text, should be a adequate method to help with web accessibility that will bypass F73.
If the link text contains more text than the actual link target, it also causes no harm, as the user, if he doesn't find the link, will aim for the link target (URL, email or phone number in the link text) and click.

This can be seen in the last example of links that should pass the F73 test, as the addition of " (rain-in-spain.html) " should be a adqeuate visible cue that this is a link to a file called rain-in-spain.html .

**Comment by mbgower** [comment]
**Draft Working Group Response**

If you do not distinguish links by colour to anyone, you do not fail [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG22/Understanding/use-of-color). If you do not distinguish links by any visual indicator, you are no failing [1.4.11 Non-text Contrast](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast).
But you would subject your users to a frustrating experience of having to try different pieces of text all over the screen to see if they are actionable.

In a word, with such a design, you have entered into the broad topic of usability.

The Web Content Accessibility Guidelines exist to improve the likelihood that a site usable to typical users is also usable by atypical users. As long as your site is equally unusable to everyone, you do not fail WCAG.

However, if you create links that have subtle visual differences from surrounding text, you are disadvantaging some users with low vision, and you are likely to fail one of the two criteria mentioned in this response.

**Comment by mbgower** [comment]
Take a look at https://codepen.io/Giacomo-Petri/pen/PwYwdoK

**Comment by mbgower** [comment]
Based on a review by the working group, the draft response has been changed to the [official response](https://github.com/w3c/wcag/issues/4143#issuecomment-2508066739).
